Examples/E1/CargoService.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Examples.E1
+{
+    public class CargoService
+    {
+        private  CargoFactory CargoFactory { get; set; }
+        private  string Origin =string.Empty;
+        private  string Destination = string.Empty;
+        public ICargo BookProduct(string type)
+        {
+
+            return CargoFactory.CreateCargoFactory(type,Origin,Destination);
```

Comment:
null pointer exception might be accrue here. if you forgot to create an instance for CargoFactory.